include/s3select_oper.h

```diff
@@ -775,11 +775,6 @@ class value
       return *timestamp() == *(v.timestamp());
     }
 
-    if ((is_null() || v.is_null()) || (is_nan() || v.is_nan()))
```

Comment:
why, to remove that?

any compare with NULL is UNKNOWN 
as for NAN the comparison table in docs, contain expected results (ONLY true/false)

all compare operator return bool, which is not enough, it need for third state (UNKNOWN)